sdk/resourcemanager/azure-resourcemanager-cdn/src/main/java/com/azure/resourcemanager/cdn/implementation/CdnEndpointImpl.java

```diff
@@ -115,6 +137,23 @@ public Mono<CdnEndpoint> updateResourceAsync() {
                 .withQueryStringCachingBehavior(this.innerModel().queryStringCachingBehavior())
                 .withTags(this.innerModel().tags());
 
+        if (isStandardMicrosoftSku()) {
+            List<DeliveryRule> rules = this.standardRulesEngineRuleMap.values()
+                .stream()
+                .sorted(Comparator.comparingInt(DeliveryRule::order))
+                .collect(Collectors.toList());
+
+            if (innerModel().deliveryPolicy() == null && !CoreUtils.isNullOrEmpty(this.standardRulesEngineRuleMap)) {
+                endpointUpdateParameters.withDeliveryPolicy(
+                    new EndpointPropertiesUpdateParametersDeliveryPolicy()
+                        .withRules(rules));
+            } else if (innerModel().deliveryPolicy() != null) {
+                endpointUpdateParameters.withDeliveryPolicy(
+                    innerModel().deliveryPolicy()
+                        .withRules(rules));
+            }
```

Comment:
What is this "else if" part used for?

I assume `innerModel().deliveryPolicy()` be not empty would mean `standardRulesEngineRuleMap` be not empty (if user not modify it)?

On the other hand, if `standardRulesEngineRuleMap` be empty, does it mean user actually removed the rules? Hence it need to be handled in "if" clause?

Reply:
Got it. We can always replace deliveryPolicy with `standardRulesEngineRuleMap`.